tests/contrib/algoliasearch/test.py

```diff
@@ -0,0 +1,117 @@
+import algoliasearch
+import algoliasearch.index as index_module
+from ddtrace import config
+from ddtrace.contrib.algoliasearch.patch import (SEARCH_SPAN_TYPE, patch,
+                                                 unpatch)
+from ddtrace.pin import Pin
+from tests.base import BaseTracerTestCase
+
+
+class AlgoliasearchTest(BaseTracerTestCase):
+    def setUp(self):
+        super(AlgoliasearchTest, self).setUp()
+
+        # dummy values
+        def search(self, query, args=None, request_parameters=None):
+            return {
+                'hits': [
+                    {
+                        'dummy': 'dummy'
+                    }
+                ],
+                'processingTimeMS': 23,
+                'nbHits': 1,
+                'hitsPerPage': 20,
+                'exhaustiveNbHits': True,
+                'params': 'query=xxx',
+                'nbPages': 1,
+                'query': 'xxx',
+                'page': 0
+            }
+
+        # Algolia search is a non free SaaS application, it isn't possible to add it to the
```

Comment:
We could use the `mock` library instead to ensure we don't make any outbound requests.

https://docs.python.org/3/library/unittest.mock.html

I haven't tested, but something like this _should_ do it:

```python
def setUp(self):
    super(...).setUp()

    self.index_patcher = mock.patch('algoliasearch.index.Index', spec=True)
    self.index_patcher.start()

    self.client = algoliasearch.algoliasearch.Client('X', 'X')
    self.index = self.client.init_index('test_index')
    Pin.override(self.index, tracer=self.tracer)

def tearDown(self):
    super(...).tearDown()
    self.index_patcher.stop()
```